Create map.md, closes #3068 (#3264)
Codecademy/docs: content/go/concepts/strings/terms/map/map.md

---
Title: 'Map()'
Description: 'Returns a copy of the provided string with its characters converted according to the mapping function.'
Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Functions'
CatalogContent:
  - 'learn-go'
  - 'paths/computer-science'
---

The **`strings.Map()`** method changes all of the characters of a given string according to the changes specified in the given mapping function.

## Syntax

```pseudo

> Map(mappingFunct(r),  s)

Reviewer: ```suggestion
strings.Map(mappingFunct(r), s)
```

```

The `strings.Map()` function takes two parameters:

> - `mappingFunct(r)`: A function and a character (type rune) that the original character will be replaced with.

Reviewer: We don't have an entry on runes so it would be helpful here to making a differentiation here

Reviewer: edit: I'm going to link the datatypes entry here

Author: Thank you for adding the link!

> - `funct(rune) rune`: The new character that the initial character will be replaced with.

Reviewer 2: ```suggestion
- `mappingFunct(r)`: A function and a character (type rune) that the original character will be replaced with.
```

- `s`: The given string in which the characters will be replaced.

> **Note:** If a negative value is returned when a character is mapped, it is dropped from the string without a replacement.

## Example

The following example shows the implementation of `strings.Map()`:

```go

>   package main

Reviewer: ```suggestion
package main
```

import (
  "fmt"
  "strings"
)

func main() {
  mapped := func(r rune) rune {

>     if r == "e" {

Reviewer: I think you meant to use single quotes here for runes

```suggestion
    if r == 'e' {
```
